tests/network/l2_bridge/libl2bridge.py

```
@@ -236,18 +258,82 @@ def hot_plug_interface_and_set_address(
+
+    Returns:
+        A ResourceField with name and interfaceName fields.
+
+    Raises:
+        VMInterfaceStatusNotFoundError: If no interface with the expected MAC is found.
+    """
+    cmd = "ip -j addr show"
+    output = vm_console_run_commands(vm=vm, commands=[cmd], timeout=30)
+    guest_interfaces = json.loads(output[cmd][1])
+    visible_ifaces = [{"ifname": iface.get("ifname"), "address": iface.get("address")} for iface in guest_interfaces]
+    LOGGER.info(f"CNV-77961: looking for MAC {expected_mac} in guest {vm.name}, visible interfaces: {visible_ifaces}")
```

Ruff (0.15.20)

[warning] Logging statement uses f-string

(G004)